pkg/apis/apps.kubermatic/v1/application_version.go

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	semverlib "github.com/Masterminds/semver/v3"
+)
+
+// +kubebuilder:validation:Type=string
+
+// Version wraps semverlib.Version. It is needed because kubebuilder does not accept structs with non-tagged fields, even if they have custom marshallers
+// With this the CRD resource will have Version as string but operator code can work directly with the semverlib.Version struct
+// (taken from https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/crd/testdata/cronjob_types.go#L283)
+type Version struct {
```

Comment:
What about using pkg/semver ?

Reply:
I have never looked too deeply into our semver package. It existed since forever and just mostly works ^^